vault/router.go

```diff
@@ -174,6 +181,31 @@ func (r *Router) MatchingMount(path string) string {
 	return mount
 }
 
+//MatchingPrefix returns a mount prefix that a path may be a part of
+func (r *Router) MatchingPrefix(path string) string {
+	var existing string = ""
+	fn := func(existing_path string, _v interface{}) bool {
```

Comment:
If this function executes _at all_ wouldn't it mean that we've found a prefix? Is the `if` check unnecessary here?